cadl-tests/src/main/java/com/cadl/optional/models/WithOptional.java

```diff
@@ -0,0 +1,381 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.cadl.optional.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.CoreUtils;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+
+/** The WithOptional model. */
+@Fluent
+public final class WithOptional {
+    /*
+     * The boolean property.
+     */
+    @JsonProperty(value = "boolean")
+    private Boolean booleanProperty;
+
+    /*
+     * The booleanNullable property.
+     */
+    @JsonProperty(value = "booleanNullable")
+    private Boolean booleanNullable;
+
+    /*
+     * The booleanRequired property.
+     */
+    @JsonProperty(value = "booleanRequired")
+    private Boolean booleanRequired;
```

Comment:
what does `withOptionalProperties` do? why `booleanRequired` here?

Reply:
It makes all the properties in a model optional: https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#withoptionalproperties

`WithOptional` has the spread properties of `Optional` and `booleanRequired` is defined in `Optional` as a required property. The goal here is to test whether `withOptionalProperties` works as intended.

Reply:
I see.

These wasn't there in tutorial last time...

Reply:
Yes.. And it has different semantics from `withVisibility` that haoling did test on, which acts like as a filter to filter out properties that don't conform with the visibility specified.

Renamed it to `AllPropertiesOptional` to avoid confusion.